core/src/main/java/com/novoda/noplayer/internal/exoplayer/NoPlayerAdsLoader.java

```diff
@@ -327,26 +303,10 @@ void enableAdverts() {
             return;
         }
 
-        long contentPosition = player.getContentPosition();
-        AvailableAdverts.markAllFutureAdvertsAsAvailable(contentPosition, advertBreaks, adPlaybackState);
+        AvailableAdverts.markSkippedAdvertsAsAvailable(advertBreaks, adPlaybackState);
 
         updateAdPlaybackState();
         advertListener.onAdvertsEnabled(advertBreaks);
         advertsDisabled = false;
     }
-
-    @Override
-    public void onBeat(NoPlayer noPlayer) {
-        if (player == null || player.isPlayingAd()) {
-            return;
-        }
-
-        lastContentPositionInMillis = player.getContentPosition();
-
-        if (hasPerformedSeek) {
```

Comment:
We aren't concerned with disabling adverts on seek now, they can play based on the state they are currently in.